scanpipe/pipelines/docker.py

```diff
@@ -77,21 +83,6 @@ def collect_images_information(self):
         images_data = [docker.get_image_data(image) for image in self.images]
         self.project.update_extra_data({"images": images_data})
 
-    def collect_and_create_codebase_resources(self):
-        """
-        Collects and labels all image files as CodebaseResources.
-        """
-        for image in self.images:
-            docker.create_codebase_resources(self.project, image)
-
-    def collect_and_create_system_packages(self):
-        """
-        Collects installed system packages for each layer based on the distro.
-        """
-        with self.save_errors(rootfs.DistroNotFound, rootfs.DistroNotSupported):
-            for image in self.images:
-                docker.scan_image_for_system_packages(self.project, image)
-
```

Comment:
IMHO you cannot get rid of `collect_and_create_codebase_resources()` as a first step and and of `collect_and_create_system_packages()` when scanning a docker image.
Instead we need to find a way to continue to scan by image and in each image by layer otherwise things will be incorrect.